### PR TITLE
Added tests, and test files for WMC (Warn on missing contig) option.

### PR DESCRIPTION
**Pull request by fleharty**
@yfarjoun Since you know what this is about, you might be a good candidate for review.

### CODE REVIEW

File: src/tests/java/picard/vcf/LiftoverVcfTest.java [hunk 1]
```diff
@@ -92,4 +93,37 @@ public void testFixReverseComplementedGenotypes() {
...
+        // Test without using WMC option (Should exit with EXIT_CODE_WHEN_CONTIG_NOT_IN_REFERENCE)
+        final String[] argsWithoutWarnOnMissingContig = new String[]{
+                "INPUT=" + input.getAbsolutePath(),
+                "OUTPUT=" + liftOutputFile.getAbsolutePath(),
+                "REJECT=" + rejectOutputFile.getAbsolutePath(),
+                "CHAIN=" + CHAIN_FILE_WITH_BAD_CONTIG,
+                "REFERENCE_SEQUENCE=" + REFERENCE_FILE,
+                "CREATE_INDEX=false"
+        };
+        Assert.assertEquals(runPicardCommandLine(argsWithoutWarnOnMissingContig), LiftoverVcf.EXIT_CODE_WHEN_CONTIG_NOT_IN_REFERENCE);
```
Comment (yfarjoun) [inline src/tests/java/picard/vcf/LiftoverVcfTest.java]: break this up into a test and a dataprovider

File: src/java/picard/vcf/LiftoverVcf.java [hunk 2]
```diff
@@ -85,6 +85,12 @@
                     "accompanying sequence dictionary (.dict file).")
     public File REFERENCE_SEQUENCE = Defaults.REFERENCE_FASTA;
 
+    @Option(shortName = "WMC", doc = "Warn on missing contig.", optional = true)
```
Comment (yfarjoun) [inline src/java/picard/vcf/LiftoverVcf.java]: long name should be descriptive: WARN_ON_MISSING_CONTIG.

as opposed to what? what happens if this is false?

File: src/java/picard/vcf/LiftoverVcf.java [hunk 3]
```diff
@@ -85,6 +85,12 @@
...
+    public boolean WMC = false;
+
+    @Option(doc="When a contig used in the chain is not in the reference, exit with this value instead of 0.")
```
Comment (yfarjoun) [inline src/java/picard/vcf/LiftoverVcf.java]: I don't think this should be an input option. make it a protected (for testing) static value.

File: src/java/picard/vcf/LiftoverVcf.java [hunk 4]
```diff
@@ -174,6 +180,21 @@ public static void main(final String[] args) {
...
+            else if (refSeqs.get(target.getContig()) == null) {
+                final String reason = FILTER_NO_TARGET;
+                rejects.add(new VariantContextBuilder(ctx).filter(reason).make());
+                failedLiftover++;
+
+                String missingContigMessage = "Encountered a contig, " + target.getContig() + " that is not part of the reference.";
+                if(WMC) {
+                    log.warn(missingContigMessage);
+                }
+
```
Comment (yfarjoun) [inline src/java/picard/vcf/LiftoverVcf.java]: use 
```code
} else {  
```
(all in the same line)

File: src/java/picard/vcf/LiftoverVcf.java [hunk 5]
```diff
@@ -174,6 +180,21 @@ public static void main(final String[] args) {
...
+                else {
+                    log.error(missingContigMessage);
+                    return EXIT_CODE_WHEN_CONTIG_NOT_IN_REFERENCE;
+                }
+            }
```
Comment (yfarjoun) [inline src/java/picard/vcf/LiftoverVcf.java]: here too } else {

File: src/java/picard/vcf/LiftoverVcf.java [hunk 6]
```diff
@@ -174,6 +180,21 @@ public static void main(final String[] args) {
                 rejects.add(new VariantContextBuilder(ctx).filter(reason).make());
                 failedLiftover++;
             }
+            else if (refSeqs.get(target.getContig()) == null) {
+                final String reason = FILTER_NO_TARGET;
+                rejects.add(new VariantContextBuilder(ctx).filter(reason).make());
+                failedLiftover++;
+
+                String missingContigMessage = "Encountered a contig, " + target.getContig() + " that is not part of the reference.";
```
Comment (yfarjoun) [inline src/java/picard/vcf/LiftoverVcf.java]: reference -> target reference

File: src/java/picard/vcf/LiftoverVcf.java [hunk 7]
```diff
@@ -174,6 +180,21 @@ public static void main(final String[] args) {
                 rejects.add(new VariantContextBuilder(ctx).filter(reason).make());
                 failedLiftover++;
             }
+            else if (refSeqs.get(target.getContig()) == null) {
```
Comment (yfarjoun) [inline src/java/picard/vcf/LiftoverVcf.java]: use has() rather than get() == null.

**Comment by yfarjoun** [comment]
What about the same protection for LiftOverIntervalList? there you'll see the same problem:

the add() in line 99 will throw an IllegalArgumentException if the contig isn't in the dictionary.

**Comment by yfarjoun** [comment]
something about the commits here is wrong...this needs some careful rebaseing.

Please do not merge as is!!!

@fleharty please rebase carefully.

**Comment by fleharty** [comment]
@yfarjoun I noticed this earlier so I rebased this into a different branch that you rereviewed, and then I merged in.  I'm going to delete this branch and pull request.

**Comment by yfarjoun** [comment]
no problem.

**Pull request closed without merging**